src/lambda/vue/samInvoke.ts

```diff
@@ -0,0 +1,501 @@
+        pickerItems.push({
+            index: -1,
+            label: localize('AWS.lambda.form.noDebugConfigs', 'No aws-sam debug configurations found.'),
+            detail: localize('AWS.picker.dynamic.noItemsFound.detail', 'Click here to go back'),
+        })
+    }
+    const qp = picker.createQuickPick({
+        items: pickerItems,
+        options: {
+            title: localize('AWS.lambda.form.selectDebugConfig', 'Select Debug Configuration'),
+        },
+    })
+
+    const choices = await picker.promptUser({
+        picker: qp,
+    })
+    const pickerResponse = picker.verifySinglePickerOutput<LaunchConfigPickItem>(choices)
+
+    if (!pickerResponse || pickerResponse.index === -1) {
+        return
+    }
+    postMessageFn({
+        command: 'loadSamLaunchConfig',
+        data: {
+            launchConfig: pickerResponse.config!,
+        },
+    })
+}
+
+interface SampleRequestManifest {
+    requests: {
+        request: {
+            name?: string
+            filename?: string
+        }[]
+    }
+}
+
+interface SampleQuickPickItem extends vscode.QuickPickItem {
+    filename: string
+}
+
+/**
+ * Open a quick pick containing upstream sample payloads.
+ * Call back into the webview with the contents of the payload to add to the JSON field.
+ * @param postMessageFn
+ */
+async function getSamplePayload(postMessageFn: (response: SamInvokerResponse) => Thenable<boolean>): Promise<void> {
+    // stolen from invokeLambda.ts
+    try {
+        const sampleInput = await makeSampleRequestManifestResourceFetcher().get()
+
+        if (!sampleInput) {
+            throw new Error('Unable to retrieve Sample Request manifest')
+        }
+
+        getLogger().debug(`Loaded: ${sampleInput}`)
+
+        const inputs: SampleQuickPickItem[] = []
+
+        await new Promise<void>((resolve, reject) => {
+            xml2js.parseString(sampleInput, { explicitArray: false }, (err: Error, result: SampleRequestManifest) => {
+                if (err) {
+                    reject()
+                }
+
+                _.forEach(result.requests.request, r => {
+                    inputs.push({ label: r.name ?? '', filename: r.filename ?? '' })
+                })
+                resolve()
+            })
+        })
+
+        const qp = picker.createQuickPick({
+            items: inputs,
+            options: {
+                title: localize('AWS.lambda.form.pickSampleInput', 'Pick a Sample Input'),
+            },
+        })
+
+        const choices = await picker.promptUser({
+            picker: qp,
+        })
+        const pickerResponse = picker.verifySinglePickerOutput<SampleQuickPickItem>(choices)
+
+        if (!pickerResponse) {
+            return
+        }
+        const sampleUrl = `${sampleRequestPath}${pickerResponse.filename}`
+        const sample = (await new HttpResourceFetcher(sampleUrl, { showUrl: true }).get()) ?? ''
+        // declaring here so we don't get an error
+        sample
```

Comment:
is this still needed? It is already declared in the `const sample` , and it is referenced in `payload: sample` below.

Reply:
I think this was hackathon leftovers/copy-paste. Will take a look.